src/mcts/helpers.rs

```diff
@@ -60,35 +60,55 @@ impl SearchHelpers {
     /// This will be overriden by a `go movetime` command,
     /// and a move overhead will be applied to this, so no
     /// need for it here.
-    pub fn get_time(time: u64, increment: Option<u64>, ply: u32, movestogo: Option<u64>) -> u128 {
+    pub fn get_time(
+        time: u64,
+        increment: Option<u64>,
+        ply: u32,
+        movestogo: Option<u64>,
+        params: &MctsParams,
+    ) -> (u128, u128) {
         let inc = increment.unwrap_or(0);
 
-        let mut max_time = if let Some(mtg) = movestogo {
+        let opt_time;
+        let max_time;
```

Comment:
You don't need to do this, in the block below you can just directly return the tuple from the function:

```rust
        if let Some(mtg) = movestogo {
            // Cyclic time control (x moves in y seconds)
            let time = (time as f64 / (mtg as f64).clamp(1.0, 30.0)) as u128;
            (time, time)
        } else {
            // Increment time control (x seconds + y increment)
            let mtg = 30;

            // blah blah ...

            let opt_time = (opt_scale * bonus * time_left) as u128;
            let max_time =
                (max_scale * opt_time as f64).min(time as f64 * params.tm_max_time()) as u128;

            (opt_time, max_time)
        }
```